Added domain and repositories

caramandvd opened the pull request:
Acum lucrez la functia "_add user_" to db.

caramandvd committed:
am facut partea de read all si read single user by user_id

caramandvd committed:
Am REFACUT partea de Domain si am facut UsersRespository.

caramandvd committed:
Am REFACUT partea de Domain si am facut UsersRespository.

caramandvd committed:
Am facut niste modificari la user_repositories.py si am facut user_account_repo

caramandvd committed:
Am facut users_transaction_repository.py

caramandvd committed:
am redenumit fisierul

EcclesiaITAcademy commented on Utils/utils.py:
```diff
@@ -1,29 +1,17 @@
+engine = create_engine('mysql+pymysql://root@localhost:3306/bankdashboarddb')
+
+
+
```
Nitpick: You can remove this extra spaces

EcclesiaITAcademy commented on Utils/utils.py:
```diff
@@ -1,29 +1,17 @@
+
+
+
+# if __name__ == '__main__':
```
You don't need to comment this out. It will only run when you run utils.py directly.

caramandvd committed:
AM ADAUGAT currencies_repository.py

EcclesiaITAcademy commented on Utils/utils.py:
```diff
@@ -1,29 +1,17 @@
+
+# if __name__ == '__main__':
+#     engine = create_engine('mysql+pymysql://root@localhost:3306/bankdashboarddb')
+#     Base.metadata.create_all(engine)
```
Since in the utils you only create a connection and use it elsewhere you can just test to see if the connection works:

`
try:
    engine.connect()
    print("Success")
except SQLAlchemyError as err:
    print("Error", err.__cause__)  # this will give what kind of error
`

EcclesiaITAcademy commented on Model/Domain/atm_locations.py:
```diff
@@ -0,0 +1,11 @@
+from Utils.utils import Base
+from sqlalchemy import Column, String, Integer, Date, Float, ForeignKey, Double
```
Try to import only the things we're using (Date, Float, ForeignKey, Double not used but imported)

EcclesiaITAcademy commented on Model/Domain/currencies.py:
```diff
@@ -0,0 +1,14 @@
+from Utils.utils import Base
+from sqlalchemy import Column, String, Integer, Date, Float, ForeignKey, Double
```
Same here

EcclesiaITAcademy commented on Model/Domain/exchange_rates.py:
```diff
@@ -0,0 +1,16 @@
+from Utils.utils import Base
+from sqlalchemy import Column, String, Integer, DateTime, Float, ForeignKey, Double
```
Same

EcclesiaITAcademy commented on Model/Domain/exchange_rates.py:
```diff
@@ -0,0 +1,16 @@
+    to_currency = Column(String(3))
+    date_time = Column(DateTime)
+    rate = Column(Float(2))
+    def __repr__(self):
```
Nitpick: An extra line above functions looks better. 

Pro tip: If it's underlined you can press "Alt" + "Enter" for suggestions. In this case "Reformat file" works best 
Also PyCharm has a "Problems" tab in the lower left that will tell you all the errors and warnings in the file

EcclesiaITAcademy commented on Model/Domain/exchange_rates.py:
```diff
@@ -0,0 +1,16 @@
+class ExchangeRates(Base):
+    __tablename__ = "exchange_rates"
+    id = Column(Integer, primary_key=True)
+    from_currency = Column(String(3))
```
A foreign key relationship is needed here

Something like this should work:

```
from sqlalchemy.orm import relationship

class Currencies(Base):
    __tablename__ = "currencies"
    currency = Column(String(3), primary_key=True)
    from_currency = relationship("ExchangeRates", backref="currencies")
    to_currency = relationship("ExchangeRates", backref="currencies")


class ExchangeRates(Base):
    __tablename__ = "exchange_rates"
    id = Column(Integer, primary_key=True)
    from_currency = Column(String(3), ForeignKey('currencies.id'))
    to_currency = Column(String(3), ForeignKey('currencies.id'))
    date_time = Column(DateTime)
    rate = Column(Float(2))

```

caramandvd replied:
added in currencies.py the requested changes. @EcclesiaITAcademy, I need you to check if I did it right. Thank!

EcclesiaITAcademy commented on Model/Domain/messages.py:
```diff
@@ -0,0 +1,17 @@
+class Messages(Base):
+    __tablename__ = 'messages'
+    id = Column(Integer, primary_key=True)
+    user_id = Column(String(13), primary_key=True, foreign_key=ForeignKey('users.user_id', ondelete="CASCADE"))
```
Add Foreign Key for user_id

EcclesiaITAcademy commented on Model/Domain/users_accounts.py:
```diff
@@ -0,0 +1,18 @@
+class UsersAccounts(Base):
+
+    __tablename__ = "users_accounts"
+    user_id = Column(String(13), foreign_key=ForeignKey('users.user_id', ondelete="CASCADE"))
```
Foreign Key

EcclesiaITAcademy commented on Model/Domain/users_cards.py:
```diff
@@ -0,0 +1,16 @@
+
+class UsersCards(Base):
+    __tablename__ = 'users_cards'
+    user_id = Column(String(13), foreign_key=ForeignKey('users.user_id', ondelete="CASCADE"))
```
Foreign Key

EcclesiaITAcademy commented on Model/Domain/users_credentials.py:
```diff
@@ -0,0 +1,16 @@
+class UsersCredentials(Base):
+    __tablename__ = "users_credentials"
+
+    user_id = Column(String(13), primary_key=True, foreign_key=ForeignKey("users.user_id", ondelete="CASCADE"))
```
Foreign Key

EcclesiaITAcademy commented on Model/Domain/users_deposits.py:
```diff
@@ -0,0 +1,17 @@
+
+class UsersDeposits(Base):
+    __tablename__ = 'users_deposits'
+    user_id = Column(String(13), primary_key=True, foreign_key=ForeignKey("users.user_id", ondelete="CASCADE"))
```
Foreign Key

EcclesiaITAcademy commented on Model/Domain/users_deposits.py:
```diff
@@ -0,0 +1,17 @@
+class UsersDeposits(Base):
+    __tablename__ = 'users_deposits'
+    user_id = Column(String(13), primary_key=True, foreign_key=ForeignKey("users.user_id", ondelete="CASCADE"))
+    currency = Column(String(3), foreign_key=ForeignKey('currencies.currency', ondelete="CASCADE"))
```
Foreign Key

EcclesiaITAcademy commented on Model/Domain/users_transaction.py:
```diff
@@ -0,0 +1,22 @@
+    __tablename__ = "users_transactions"
+
+    transaction_id = Column(Integer, primary_key=True, autoincrement=True)
+    user_id = Column(String(13), foreign_key=ForeignKey("users.user_id", ondelete="CASCADE"), autoincrement=True)
```
Foreign Key

EcclesiaITAcademy commented on Model/Domain/users_transaction.py:
```diff
@@ -0,0 +1,22 @@
+
+    transaction_id = Column(Integer, primary_key=True, autoincrement=True)
+    user_id = Column(String(13), foreign_key=ForeignKey("users.user_id", ondelete="CASCADE"), autoincrement=True)
+    currency = Column(String(3), foreign_key=ForeignKey("currencies.currency", ondelete="CASCADE"))
```
Foreign Key

EcclesiaITAcademy commented on Model/Domain/users_accounts.py:
```diff
@@ -0,0 +1,18 @@
+    __tablename__ = "users_accounts"
+    user_id = Column(String(13), foreign_key=ForeignKey('users.user_id', ondelete="CASCADE"))
+    account_number = Column(String(24), primary_key=True, foreign_key=ForeignKey('users_accounts.account_number', ondelete="CASCADE"))
+    currency = Column(String(3))
```
Foreign Key

EcclesiaITAcademy commented on Model/Repository/currencies_repository.py:
```diff
@@ -0,0 +1,28 @@
+
+class CurrenciesRepository:
+    def __init__(self):
+        engine = create_engine('mysql+pymysql://root@localhost:3306/bankdashboarddb')
```
We can import the engine from utils, no need to initiate a new one everytime

EcclesiaITAcademy commented on Model/Repository/users_accounts_repository.py:
```diff
@@ -0,0 +1,36 @@
+
+class UsersAccountsRepository:
+    def __init__(self):
+        engine = create_engine('mysql+pymysql://root@localhost:3306/bankdashboarddb')
```
Same here

EcclesiaITAcademy commented on Model/Repository/users_accounts_repository.py:
```diff
@@ -0,0 +1,36 @@
+        return self.session.query(UsersAccounts).filter_by(user_id=user_id).first()
+
+    def update(self, user_id, amount):
+        if amount:
```
What happens if the amount is None? How do we know if the update reached the DB or not?

EcclesiaITAcademy commented on Model/Repository/users_repository.py:
```diff
@@ -0,0 +1,68 @@
+
+class UsersRepository:
+    def __init__(self):
+        engine = create_engine('mysql+pymysql://root@localhost:3306/bankdashboarddb')
```
Same

EcclesiaITAcademy commented on Model/Repository/users_repository.py:
```diff
@@ -0,0 +1,68 @@
+    def create(self, user_id, first_name, last_name, email_name, address, phone_number):
+
+        def get_date_of_birth(user_id):
+            centuries = {
```
Nice! Cool approach!

EcclesiaITAcademy commented on Model/Repository/users_repository.py:
```diff
@@ -0,0 +1,68 @@
+        return self.session.query(Users).filter_by(user_id=user_id).first()
+
+    def update(self, user_id, **kwargs):
+        print(kwargs)
```
It's fine for now, but usualy it's better to not commit prints or debugging code to production

EcclesiaITAcademy commented on Model/Repository/users_transaction_repository.py:
```diff
@@ -0,0 +1,36 @@
+
+class UsersTransactionsRepository:
+    def __init__(self):
+        engine = create_engine('mysql+pymysql://root@localhost:3306/bankdashboarddb')
```
Same

EcclesiaITAcademy reviewed (changes requested):
Well done! Left some suggestions and some nitpicks. Great job overall

EcclesiaITAcademy commented:
Also, let's write in English. Create a more realistic and professional experience 😄

caramandvd committed:
I have created atm_locations_repository.py and added the __repr__ method in atm_locations.py.py

caramandvd committed:
I have created exchange_rates_repository.py

caramandvd committed:
I have created messages_repository.py and I updated the __str__ and __init__ methods in messages.py.py

caramandvd committed:
nitpicks :)

caramandvd committed:
I have created the users_credentials_repository.py. I added a hashing method in the create function.

caramandvd committed:
nitpicks :))

caramandvd committed:
I removed the unused imports.

caramandvd committed:
I have added 'try: engine.connect() print("Success") except SQLAlchemyError as err: print("Error", err.__cause__)'

caramandvd committed:
I removed the unused imports

caramandvd committed:
Relationship import from sqlalchemy.orm

caramandvd committed:
Added foreign_key for currencies.currency and I deleted the unused imports

caramandvd committed:
I deleted the unused imports

caramandvd committed:
I deleted the unused imports

caramandvd committed:
I deleted the unused imports

caramandvd committed:
I deleted the unused imports

caramandvd committed:
I imported the engine from Utils.utils.py

caramandvd committed:
I imported the engine from Utils.utils.py

caramandvd committed:
I added a condition so the value of the transaction is greater than 0. 'self.session.filter(amount.value > 0).add(new_user_account)'

caramandvd committed:
I added a condition so the value of the transaction is greater than 0, but this time it works

caramandvd committed:
I added a condition so the value of the transaction is greater than 0, but this time it works

caramandvd committed:
I added an extra step so I can be sure the values got into the DB

caramandvd committed:
I added an extra step so I can be sure the values got into the DB

EcclesiaITAcademy commented:
You can press on "Resolve conversation" where you did the changes so it's easier to follow

EcclesiaITAcademy commented on Model/Domain/exchange_rates.py:
```diff
@@ -0,0 +1,25 @@
+from sqlalchemy.orm import relationship
+
+
+class Currencies(Base):
```
This goes in currencies.py

caramandvd committed:
feat: created users_cards_repository.py.py and users_deposits_repository.py

caramandvd committed:
style: nitpicks

EcclesiaITAcademy commented on Model/Repository/atm_locations_repository.py:
```diff
@@ -0,0 +1,31 @@
+class AtmLocationsRepository:
+
+    def __init__(self):
+        self.session = sessionmaker(create_engine('mysql+pymysql://root@localhost:3306/bankdashboarddb'))()
```
Use engine from utils here

EcclesiaITAcademy commented on Model/Repository/messages_repository.py:
```diff
@@ -0,0 +1,33 @@
+
+class MessagesRepository:
+    def __init__(self):
+        self.session = sessionmaker(create_engine('mysql+pymysql://root@localhost:3306/bankdashboarddb'))()
```
Use engine from utils

EcclesiaITAcademy commented on Model/Repository/users_credentials_repository.py:
```diff
@@ -0,0 +1,41 @@
+
+class UsersCredentialsRepository:
+    def __init__(self):
+        self.session = sessionmaker(create_engine('mysql+pymysql://root@localhost:3306/bankdashboarddb'))()
```
Use engine from utils

EcclesiaITAcademy commented on Model/Repository/users_transaction_repository.py:
```diff
@@ -0,0 +1,32 @@
+
+class UsersTransactionsRepository:
+    def __init__(self):
+        self.session = sessionmaker(create_engine('mysql+pymysql://root@localhost:3306/bankdashboarddb'))()
```
Use engine from utils

EcclesiaITAcademy commented:
You also need to add relationships for the foreign keys

caramandvd committed:
Update atm_locations_repository.py

caramandvd committed:
Update users_credentials_repository.py

caramandvd committed:
Update atm_locations_repository.py

caramandvd committed:
Update atm_locations_repository.py

caramandvd committed:
Update currencies_repository.py

caramandvd committed:
Update exchange_rates_repository.py

caramandvd committed:
Update messages_repository.py

caramandvd committed:
Update users_accounts_repository.py

caramandvd committed:
Update users_repository.py

caramandvd committed:
Update users_transaction_repository.py

caramandvd committed:
Update currencies.py

refactor: added relationship keys

EcclesiaITAcademy committed:
style: warnings, renames

Fixed some warnings
Renamed folders
Created InvalidUserID exception
Added .gitignore

EcclesiaITAcademy reviewed (approved):
Still need to add the relationships but we can do it on another branch
